### ISSUE TITLE
Add support for XML Schema 1.1

### ISSUE THREAD

**Issue by rjmartell** [opengeospatial/teamengine#138]
XML Schema 1.1 has been a W3C recommendation for more than three years, and the [KML 2.3 specification](http://docs.opengeospatial.org/is/12-007r2/12-007r2.html) adopted it as the normative schema language. However, TEAM engine uses an XML processor that does not support it (Apache Xerces Java 2.11, released 2010-11). While the last "beta" release of Xerces included (experimental) support for XML Schema 1.1, it has not been updated in five (!) years.

**Comment by rjmartell** [comment]
The [KML 2.x test suite](https://github.com/opengeospatial/ets-kml2) cannot be deployed at the moment since it requires an XML processor that supports XML Schema 1.1. The test suite currently uses a component built from the [xml-schema-1.1-dev branch](http://svn.apache.org/viewvc/xerces/java/branches/xml-schema-1.1-dev/) (revision: 1667115); the artifact is available from the central Maven repository:

```xml
<dependency>
  <groupId>org.opengis.cite.xerces</groupId> 
  <artifactId>xercesImpl-xsd11</artifactId> 
  <version>2.12-beta-r1667115</version> 
</dependency>
```

This library could replace `xercesImpl-2.11.0.jar` if  testing indicates that it is a compatible replacement.

**Comment by rjmartell** [comment]
No anomalies observed when validating these two KML documents:

* `https://developers.google.com/kml/documentation/KML_Samples.kml` (KML 2.2): 3 fail verdicts as expected.
* `src/test/resources/kml23/Placemark-001.xml` (KML 2.3): 2 fail verdicts as expected

**Comment by rjmartell** [comment]
These test suites ran successfully:

* CSW 3.0 (0.10)
* KML 2.x (0.4-SNAPSHOT)
* WMS 1.3 (1.15)

**Comment by rjmartell** [comment]
The teamengine-core module now uses a component built from the [`xml-schema-1.1-dev`](http://svn.apache.org/viewvc/xerces/java/branches/xml-schema-1.1-dev/) branch (revision: 1667115); this artifact was built using JDK 7 and is available from the central Maven repository:

    <dependency>
      <groupId>org.opengis.cite.xerces</groupId> 
      <artifactId>xercesImpl-xsd11</artifactId> 
      <version>2.12-beta-r1667115</version> 
    </dependency>

In order to enable checking of assertions that may be defined in an XML Schema 1.1 grammar, the XPath 2.0 processor bundled with the Eclipse Web Tools Platform (WTP) is required. Unfortunately the Xerces schema processor does not use an official release of this component; rather, it uses a build based on the [`R3_2_maintenance`](http://git.eclipse.org/c/sourceediting/webtools.sourceediting.xpath.git/?h=R3_2_maintenance) branch. This special dependency has also been published to the central repository:

    <dependency> 
      <groupId>org.opengis.cite.eclipse.webtools.sse</groupId> 
      <artifactId>org.eclipse.wst.xml.xpath2.processor</artifactId> 
      <version>1.1.5-738bb7b85d</version> 
    </dependency>

__Note__: The WTP 3.2 release included v1.1.4 of the XPath 2.0 processor, and WTP 3.3 included the v2.0.0 processor. Neither version will work with Xerces.
